### PR TITLE
Add Caching to GetAssemblyFileVersion

### PR DESCRIPTION
**Pull request by mojotheo**
*Issue #107*

https://github.com/aws/aws-sdk-net-extensions-cognito/issues/107

This PR adds caching to the AssemblyFileVersion using System.Runtime.Caching to prevent making a call to fetch the assembly everytime

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by muhammad-othman** [comment]
@mojotheo Thank you for creating this PR
I believe this will help with the issue you've linked, but using System.Runtime.Caching seems like an overkill, especially that this value will never change again and we wont be using caching anywhere else.
Can you update it to just use a static variable instead?

**Comment by mojotheo** [comment]
@muhammad-othman Updated! let me know if that looks good

**Review by muhammad-othman — APPROVED** [review]

File: src/Amazon.Extensions.CognitoAuthentication/Util/CognitoAuthHelper.cs [hunk 1]
```diff
@@ -40,9 +40,28 @@ internal static SHA256 Sha256
...
+                    AssemblyFileVersionAttribute attribute = assembly.GetCustomAttribute(typeof(AssemblyFileVersionAttribute)) as AssemblyFileVersionAttribute;
+
+                    assemblyFileVersion = attribute == null ? "Unknown" : attribute.Version;
+                }
+                
+                return assemblyFileVersion;
+            }
+        }
+        
+        private const string AssemblyFileVersionCachePath = "Cognito.AssemblyFileVersion";
```
Comment (normj) [inline src/Amazon.Extensions.CognitoAuthentication/Util/CognitoAuthHelper.cs:64]: I think this is orphan code from the previous revision. Can you delete this?

**Review by normj — CHANGES_REQUESTED** [review]

File: src/Amazon.Extensions.CognitoAuthentication/Util/CognitoAuthHelper.cs (hunk 1 above)
Comment (mojotheo) [reply src/Amazon.Extensions.CognitoAuthentication/Util/CognitoAuthHelper.cs]: Removed it

**Review by normj — APPROVED** [review]

**Review by philasmar — APPROVED** [review]

**Comment by mojotheo** [comment]
Is there a timeline on when we can expect this to go in?

**Comment by normj** [comment]
Version 2.3.1 is out with this PR. Thanks for the PR!

**Pull request merged**